index.js

```diff
@@ -17,35 +17,58 @@ const serializeError = FJS({
 })
 
 function rateLimitPlugin (fastify, opts, next) {
-  const timeWindow = typeof opts.timeWindow === 'string'
+  const globalTimeWindow = typeof opts.timeWindow === 'string'
     ? ms(opts.timeWindow)
     : typeof opts.timeWindow === 'number'
       ? opts.timeWindow
       : 1000 * 60
 
   const store = opts.redis
-    ? new RedisStore(opts.redis, timeWindow)
-    : new LocalStore(timeWindow, opts.cache)
+    ? new RedisStore(opts.redis)
+    : new LocalStore(opts.cache)
 
   const keyGenerator = typeof opts.keyGenerator === 'function'
     ? opts.keyGenerator
     : (req) => req.raw.ip
 
   const skipOnError = opts.skipOnError === true
-  const max = opts.max || 1000
   const whitelist = opts.whitelist || []
-  const after = ms(timeWindow, { long: true })
+  const rules = {}
+
+  if (opts.max) {
+    rules.global = {
+      max: opts.max,
+      timeWindow: globalTimeWindow,
+      after: ms(globalTimeWindow, { long: true })
+    }
+  }
+
+  opts.special && opts.special.length && opts.special.forEach(({ url, max, timeWindow }) => {
```

Comment:
Can you please rename this to something else? I feel `special` is too generic.

Reply:
LOL. I'm not a native speaker and i feel it's a bad name too.
OK, if you think the strategy is accepted, i'll add more in this PR.